cloud/azure/azure_rm_deployment.py

```diff
@@ -0,0 +1,646 @@
+              adminPassword: "[parameters('adminPassword')]"
+            storageProfile:
+              imageReference:
+                publisher: "[variables('imagePublisher')]"
+                offer: "[variables('imageOffer')]"
+                sku: "[parameters('ubuntuOSVersion')]"
+                version: "latest"
+              osDisk:
+                name: "osdisk"
+                vhd:
+                  uri: "[concat('http://',parameters('newStorageAccountName'),'.blob.core.windows.net/',variables('vmStorageAccountContainerName'),'/',variables('OSDiskName'),'.vhd')]"
+                caching: "ReadWrite"
+                createOption: "FromImage"
+            networkProfile:
+              networkInterfaces:
+                -
+                  id: "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]"
+            diagnosticsProfile:
+              bootDiagnostics:
+                enabled: "true"
+                storageUri: "[concat('http://',parameters('newStorageAccountName'),'.blob.core.windows.net')]"
+    parameters:
+      newStorageAccountName:
+        value: devopsclestorage
+      adminUsername:
+        value: devopscle
+      adminPassword:
+        value: Password1!
+      dnsNameForPublicIP:
+        value: devopscleazure
+'''
+
+RETURN = '''
+msg:
+  description: String indicating if the deployment was created or deleted
+  returned: always
+  type: string
+  sample: "deployment created"
+deployment:
+  description: Deployment details
+  type: dict
+  returned: always
+  sample:{
+        "group_name": "Test_Deployment",
+        "id": "/subscriptions/3f7e29ba-24e0-42f6-8d9c-5149a14bda37/resourceGroups/Test_Deployment/providers/Microsoft.Resources/deployments/ansible-arm",
+        "instances": [
+            {
+                "ips": [
+                    {
+                        "dns_settings": {
+                            "domain_name_label": "testvm9910001",
+                            "fqdn": "testvm9910001.westus.cloudapp.azure.com"
+                        },
+                        "id": "/subscriptions/3f7e29ba-24e0-42f6-8d9c-5149a14bda37/resourceGroups/Test_Deployment/providers/Microsoft.Network/publicIPAddresses/myPublicIP",
+                        "name": "myPublicIP",
+                        "public_ip": "13.91.99.232",
+                        "public_ip_allocation_method": "IPAllocationMethod.dynamic"
+                    }
+                ],
+                "vm_name": "MyUbuntuVM"
+            }
+        ],
+        "name": "ansible-arm",
+        "outputs": {
+            "hostname": {
+                "type": "String",
+                "value": "testvm9910001.westus.cloudapp.azure.com"
+            },
+            "sshCommand": {
+                "type": "String",
+                "value": "ssh chouseknecht@testvm9910001.westus.cloudapp.azure.com"
+            }
+        }
+    }
+'''
+
+import time
+import yaml
```

Comment:
pyyaml is a prerequisite of ansible, as you know. so one cannot run ansible without it installed. here is what happens when you run ansible without pyyaml installed:

```
ansible-playbook -i inventory test_deployment.yml
ERROR! Unexpected Exception: No module named yaml
the full traceback was:

Traceback (most recent call last):
  File "/Users/chouseknecht/projects/azure/ansible/bin/ansible-playbook", line 79, in <module>
    mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
  File "/Users/chouseknecht/projects/azure/ansible/lib/ansible/cli/__init__.py", line 27, in <module>
    import yaml
```
Not sure how you expect the module to trap that, but just satisfy you. I'm adding the unnecessary try/except block.